language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -879,3 +842,67 @@ impl Default for ReadWriteSetState {
         }
     }
 }
+
+// =================================================================================================
+// Converting Infer Result into standalone types
+
+impl Offset {
+    pub fn normalize(&self, env: &GlobalEnv) -> Option<RWOffset> {
+        Some(match self {
+            Offset::Field(idx) => RWOffset::Field(*idx),
+            Offset::VectorIndex => RWOffset::VectorIndex,
+            Offset::Global(s) => RWOffset::Global(s.get_type().into_struct_type(env)?),
+        })
+    }
+}
+impl AccessPath {
+    pub fn normalize(&self, env: &GlobalEnv) -> Option<Vec<RWAccessPath>> {
+        let (roots, access_opt) = match self.root() {
+            Root::Formal(idx) => (vec![RWRoot::Formal(*idx)], None),
+            Root::Global(key) => (
+                key.address()
+                    .get_concrete_addresses()
+                    .into_iter()
+                    .map(RWRoot::Const)
+                    .collect(),
+                key.struct_type().get_type().into_struct_type(env),
+            ),
+            _ => return None,
+        };
+        let mut offsets = access_opt
```

Comment:
Might be cleaner to create `offsets` at line 860 instead of introducing the indirection of `access_opt`

Reply:
Fixed!